test/unit/host_test.rb

```diff
@@ -796,6 +796,20 @@ def teardown
       assert_equal ["#{pc} does not belong to the #{h.environment} environment"], h.errors[:puppetclasses]
     end
 
+    test "when saving a host, do not require a puppet environment" do
+      h = FactoryGirl.create(:host, :environment => environments(:production), :puppetmaster => smart_proxies(:puppetmaster))
```

Comment:
Will update the tests to use build().  I'll leave 'h' there to be more consistent, but i would tend to agree with you :)